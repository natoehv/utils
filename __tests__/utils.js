'use strict';

const utils = require('../lib/utils');

/**
 * .uriBuilder()
 */

test('.uriBuilder() - no arguments - should throw', () => {
    expect(() => {
        utils.uriBuilder();
    }).toThrowError(/Invalid URL/);
});

test('.uriBuilder() - "base is empty" - should throw', () => {
    expect(() => {
        utils.uriBuilder('/podlet.html');
    }).toThrowError(/Invalid URL/);
});

test('.uriBuilder() - "base" has long path with <filename>.json - should replace <filename>.json file with "input"', () => {
    const result = utils.uriBuilder(
        '/podlet.html',
        'http://localhost:7000/podlet/a/manifest.json'
    );
    expect(result).toBe('http://localhost:7000/podlet/a/podlet.html');
});

test('.uriBuilder() - "base" has short path with <filename>.json - should replace <filename>.json file with "input"', () => {
    const result = utils.uriBuilder(
        '/podlet.html',
        'http://localhost:7000/manifest.json'
    );
    expect(result).toBe('http://localhost:7000/podlet.html');
});

test('.uriBuilder() - "base" has long path without <filename>.json - should append "input" to "base"', () => {
    const result = utils.uriBuilder(
        '/podlet.html',
        'http://localhost:7000/podlet/a/'
    );
    expect(result).toBe('http://localhost:7000/podlet/a/podlet.html');
});

test('.uriBuilder() - "base" has short path without <filename>.json - should append "input" to "base"', () => {
    const result = utils.uriBuilder('/podlet.html', 'http://localhost:7000/');
    expect(result).toBe('http://localhost:7000/podlet.html');
});

test('.uriBuilder() - "input" does not begin with "/" - should replace <filename>.json file with "input"', () => {
    const result = utils.uriBuilder(
        'podlet.html',
        'http://localhost:7000/podlet/a/manifest.json'
    );
    expect(result).toBe('http://localhost:7000/podlet/a/podlet.html');
});

test('.uriBuilder() - "base" is without <filename>.json and does not end with "/" - should append "input" to "base"', () => {
    const result = utils.uriBuilder('/podlet.html', 'http://localhost:7000');
    expect(result).toBe('http://localhost:7000/podlet.html');
});

test('.uriBuilder() - "extra" is provided - should append "extra"', () => {
    const result = utils.uriBuilder(
        '/podlet',
        'http://localhost:7000/foo/',
        '/a/b'
    );
    expect(result).toBe('http://localhost:7000/foo/podlet/a/b');
});

/**
 * .uriIsRelative()
 */

test('.uriIsRelative() - "uri" is relative - should return "true"', () => {
    expect(utils.uriIsRelative('/manifest.json')).toBe(true);
});

test('.uriIsRelative() - "uri" is absolute - should return "false"', () => {
    expect(utils.uriIsRelative('http://localhost:7000/manifest.json')).toBe(
        false
    );
});

/**
 * .uriRelativeToAbsolute()
 */

test('.uriRelativeToAbsolute() - "input" is relative - should build absolute URI', () => {
    const result = utils.uriRelativeToAbsolute(
        '/podlet',
        'http://localhost:7000/foo/',
        '/a/b'
    );
    expect(result).toBe('http://localhost:7000/foo/podlet/a/b');
});

test('.uriRelativeToAbsolute() - "input" is absolute - should return absolute URI', () => {
    const result = utils.uriRelativeToAbsolute(
        'http://localhost:7000/foo/podlet/a/b',
        'http://localhost:7000/bar/',
        '/b/a'
    );
    expect(result).toBe('http://localhost:7000/foo/podlet/a/b');
});

test('.uriRelativeToAbsolute() - no arguments - should throw', () => {
    expect.hasAssertions();

    expect(() => {
        utils.uriRelativeToAbsolute();
    }).toThrowError(/Invalid URL/);
});
