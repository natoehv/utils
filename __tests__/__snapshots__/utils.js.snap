// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.template() - arguments given - handles v4 js and css syntax 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl1.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl2.com\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl3.com\\">
        <script src=\\"http://somejsurl1.com\\" defer></script>
        <script src=\\"http://somejsurl2.com\\" defer></script>
        <script src=\\"http://somejsurl3.com\\" defer></script>
        <title></title>
        
    </head>
    <body>
        
    </body>
</html>"
`;

exports[`.template() - arguments given - should render template using values given 1`] = `
"<!doctype html>
<html lang=\\"en-NZ\\">
    <head>
        <meta charset=\\"utf-pretend-encoding\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://somecssurl.com\\">
        <script src=\\"http://somejsurl.com\\" defer></script>
        <title>this goes in the title tag</title>
        this goes in the head section
    </head>
    <body>
        this goes in the body section
    </body>
</html>"
`;

exports[`.template() - no arguments given - should render template 1`] = `
"<!doctype html>
<html lang=\\"en-US\\">
    <head>
        <meta charset=\\"utf-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=Edge\\">
        
        
        <title></title>
        
    </head>
    <body>
        
    </body>
</html>"
`;
